### PR TITLE
Cherry-pinking TASK_LOST changes from 0.15

### PR DESCRIPTION
**Pull request by meichstedt**
Don't merge yet; I've commented-out a test that has to be migrated to changed codebase.

### CODE REVIEW

File: src/main/scala/mesosphere/marathon/core/launchqueue/impl/AppTaskLauncherActor.scala [hunk 1]
```diff
@@ -349,7 +349,9 @@ private class AppTaskLauncherActor(
       sender ! MatchedTaskOps(offer.getId, Seq.empty)
 
     case ActorOfferMatcher.MatchOffer(deadline, offer) =>
-      val matchRequest = TaskOpFactory.Request(app, offer, tasksMap, tasksToLaunch)
+      import org.apache.mesos.Protos.TaskState
+      val reachableTasks = tasksMap.values.filterNot(_.mesosStatus.fold(false)(_.getState == TaskState.TASK_LOST))
```
Comment (aquamatthias) [inline src/main/scala/mesosphere/marathon/core/launchqueue/impl/AppTaskLauncherActor.scala]: Perhaps  `exists` instead of `fold` is easier to read?

**Comment by aquamatthias** [comment]
LGTM

**Comment by meichstedt** [comment]
@aquamatthias I just added the missing test case `Don't pass the task factory lost tasks when asking for new tasks` in `AppTaskLauncherActorTest` (already squashed with the related commit)

**Comment by aquamatthias** [comment]
THANKS!

**Pull request merged**